### PR TITLE
Add Prettier GitHub workflow

### LINKED ISSUE #9082: Update and automate the running of Prettier
Undo https://github.com/pantheon-systems/documentation/pull/9047/commits/4ba642056d75b77fa5afac890c789638e97e69ab based on convo in Slack

### PR DESCRIPTION
**Pull request by jazzsequence**
Fixes #9082 

## Summary

per https://github.com/pantheon-systems/documentation/pull/9163#issuecomment-2286660654 and https://github.com/pantheon-systems/documentation/pull/9163#issuecomment-2287173776:

- Adds/updates ESLint. This was added as part of the [initial implementation](https://github.com/pantheon-systems/documentation/pull/9163#issuecomment-2286497240) which has been removed. ESLint was left because we'll probably use it again in the future.
- Adds a new `npm run lint` script in `package.json`
- Updates the README.md to note how to run ESLint locally or fix all style issues with Prettier (using the existing `npm run format` command)
- Updates the Prettier config to use JS-based configuration and splits the difference between the old/existing config and the new config in https://github.com/pantheon-systems/documentation/commit/4ba642056d75b77fa5afac890c789638e97e69ab
- Adds a new [Prettier GitHub action](https://github.com/pantheon-systems/documentation/blob/9082-re-add-prettier/.github/workflows/prettier.yml) to run the linter when any `*.js` file in `src/` is affected by a pull request and _only_ on the changed files. 
  - Prettier-formatted files affected by this are then committed back to the repository by the robot and a comment is left in the PR that files were changed (see https://github.com/pantheon-systems/documentation/pull/9180#issuecomment-2292403319 for an example)

### CODE REVIEW

**Review by stevector — CHANGES_REQUESTED** [review]
The prettier docs discourage using `eslint-plugin-prettier` as the method of executing the prettier check: https://prettier.io/docs/en/integrating-with-linters.html

Here's a summary video: https://www.youtube.com/watch?v=Cd-gBxzcsdA

I haven't yet watched the longer talk: https://www.youtube.com/watch?v=sSJBeWPIipQ&t=0s

I _think_ we should first configure prettier in isolation, and secondarily configure ESLint.

**Comment by jazzsequence** [comment]
When I dug into Prettier, I discovered that the _whole point_ of it is to edit files. This puts it at odds with [reporting that there are linting issues](https://github.com/pantheon-systems/documentation/issues/9082#issuecomment-2276541994) as opposed to _fixing_ those linting issues. ESLint is a linter and, in this context, it uses the ruleset established by Prettier as the rules it checks against. 

I can remove ESLint, but then we're back to _changing large numbers of files_ (at least the first time it runs) rather than simply _reporting that there are problems_. I could just not commit those changes back to the PR, and report a failure if files were changed by Prettier, but it seems like that maybe defeats the purpose of the tool.

**Comment by jazzsequence** [comment]
New idea:

* remove ESLint
* run Prettier on PRs. it will change files. the GH action branch will now be ahead of the PR branch.
* run a git diff against the changes Prettier made
* post a comment back on the PR saying "these files have linting issues"
* (optional) maybe output the actual diff in the PR comment as well?

**Comment by stevector** [comment]
I link that idea, and I'd be fine with the same flow for ESLint, I think we can start with just prettier though (up to you if you'd rather remove ESLint or keep it and add a separate action for it (or add on to the same Action)).

Back to my comment here: https://github.com/pantheon-systems/documentation/issues/9082#issuecomment-2276541994 I was initially hesitant about CI making the change/commit for us because I feared it'd supplant a learning opportunity. But, I see the potential for learning in this flow
* I'm doing development work in a branch
* I push/make a PR
* CI finds and fixes whitespace/etc with a new commit
* I push again
* Push rejected because of conflicts
* I pull and learn

**Comment by jazzsequence** [comment]
swinging back around to this. 

1. Let's keep ESLint. We'll probably use it in the future.
2. Let's pull the ESLint _action_ out.
3. Let's add a Prettier action that runs on changed files in PRs
  - If files were changed, it will commit the change back to the PR.
  - If it commits back to the PR, it posts a message back to the PR that looks like this: https://github.com/pantheon-systems/wordpress-composer-managed/pull/151#issuecomment-2265620483
4. ESLint checks can come _after_ this workflow. In that way, ESLint would check all the things that Prettier couldn't fix automatically, as well as checking _already fixed_ code.

**Comment by jazzsequence** [comment]
Example robot comment and change: https://github.com/pantheon-systems/documentation/pull/9180#issuecomment-2292397039

File: eslint.config.js
Comment (stevector) [inline eslint.config.js:1]: @jazzsequence I'm still concerned with intermingling prettier and eslint: https://github.com/pantheon-systems/documentation/pull/9163#pullrequestreview-2235602505

I'm fine with having both, and even adding both in the same PR, but understanding is still that it is better to use Prettier directly than having ESLint be the mechanism for running Prettier.

**Review by stevector — CHANGES_REQUESTED** [review]

File: eslint.config.js
Comment (stevector) [reply eslint.config.js:1]: Flerp, or maybe I just wasn't looking at the substance of this PR closely enough and it is doing what we want. Although, I'm going to suggest some readme tweaks.

File: README.md [hunk 1]
```diff
@@ -110,6 +110,20 @@ lando start
 
 You can view the local environment at `localhost:8000/`. Updates to docs are automatically refreshed in the browser.
 
+## Linting and Code Formatting
+We use ESLint and Prettier to enforce code style. On each pull request to the repository, if any `.js`, `.jsx`, `.ts` or `.tsx` files are modified in the `/src` directory, ESLint will run against our Prettier configuration to check for code styling issues on the updated/changed files.
```
Comment (stevector) [inline README.md:114]: > ESLint will run against our Prettier configuration to check for code styling issues on the updated/changed files.

I don't think this is accurate. The command run is `npx prettier` (which I think is the command we want)

File: package.json [hunk 2]
```diff
@@ -64,10 +64,14 @@
     "tocbot": "^4.20.0"
   },
   "devDependencies": {
+    "@eslint/eslintrc": "^3.1.0",
     "@octokit/rest": "^16.43.2",
     "date-fns": "^2.29.3",
+    "eslint": "^9.8.0",
+    "eslint-config-prettier": "^9.1.0",
+    "eslint-plugin-prettier": "^5.2.1",
```
Comment (stevector) [inline package.json:72]: I don't think we need plugin-prettier. Just config-prettier

File: eslint.config.js
Comment (jazzsequence) [reply eslint.config.js:1]: 👍

**Comment by jazzsequence** [comment]
Made the requested changes and the Prettier workflow still seems to be working as expected (see https://github.com/pantheon-systems/documentation/pull/9180#issuecomment-2408033041). 

I'm going to close that other test PR since I think I'm done with the testing, this should be good for you to use @stevector.

File: README.md [hunk 3]
```diff
@@ -139,6 +139,20 @@ lando start
...
+To check for linting issues locally, run:
+```bash
+npm run lint
+```
+
+To automatically fix formatting issues across the entire `/src` directory, run:
+```bash
+npm run format
+```
+Be cautious when running this command, as it will automatically fix any linting issues it can.
```
Comment (stevector) [inline README.md:154]:
```suggestion
Be cautious when running this command, as it will automatically fix any formatting issues it can.
```

**Review by stevector — APPROVED** [review]

**Pull request merged**